pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
Hmm - actually I'm not convinved about what @lavalamp suggested.

If you are setting RV in your watch you really expect to see only events with greater RV. It's pretty misleading that RV of the "deletion event" in fact is in the past.

So I think that the problem is completely different here. Basically the problem is that ResourceVersion of watch even is something completely different than ResourceVersion of the object in this event.

So I think that long term, we should separate those two values.

However, I think that short term I would change the RV of deleted object to be "now" (which is the initial proposal of @hongchaodeng ). I think both solutions are misleading, but this one is less misleading IMHO.